Code/Framework/AzToolsFramework/Tests/Prefab/PrefabDetachPrefabTests.cpp

```diff
@@ -0,0 +1,256 @@
+        AZ::ComponentApplicationBus::BroadcastResult(
+            childEntityName, &AZ::ComponentApplicationRequests::GetEntityName, entityOrderArrayBeforeDetach[2]);
+        EXPECT_EQ(childEntityName, houseEntityName);
+
+        // Detach the car prefab.
+        PrefabOperationResult result = m_prefabPublicInterface->DetachPrefab(carContainerId);
+        ASSERT_TRUE(result.IsSuccess());
+
+        PropagateAllTemplateChanges();
+
+        // Validate child entity order after detaching the car prefab.
+        AzToolsFramework::EntityOrderArray entityOrderArrayAfterDetach =
+            AzToolsFramework::GetEntityChildOrder(GetRootContainerEntityId());
+        EXPECT_EQ(entityOrderArrayAfterDetach.size(), 3);
+
+        childEntityName = "";
+        AZ::ComponentApplicationBus::BroadcastResult(
+            childEntityName, &AZ::ComponentApplicationRequests::GetEntityName, entityOrderArrayAfterDetach[0]);
+        EXPECT_EQ(childEntityName, stationEntityName);
+        AZ::ComponentApplicationBus::BroadcastResult(
+            childEntityName, &AZ::ComponentApplicationRequests::GetEntityName, entityOrderArrayAfterDetach[1]);
+        EXPECT_EQ(childEntityName, carPrefabName);
+        AZ::ComponentApplicationBus::BroadcastResult(
+            childEntityName, &AZ::ComponentApplicationRequests::GetEntityName, entityOrderArrayAfterDetach[2]);
+        EXPECT_EQ(childEntityName, houseEntityName);
+    }
+
+    TEST_F(PrefabDetachPrefabTests, DetachPrefabValidatesDetachedChildEntityOrder)
+    {
+        // Validate the sort order of top-level child entities.
+        //
+        // Level
+        // | Car       (prefab)  <-- detach prefab
+        //   | Engine
+        //   | Wheel   (prefab)
+        //     | Tire
+        //   | Battery
+
+        const AZStd::string carPrefabName = "CarPrefab";
+        const AZStd::string wheelPrefabName = "WheelPrefab";
+
+        const AZStd::string tireEntityName = "Tire";
+        const AZStd::string engineEntityName = "Engine";
+        const AZStd::string batteryEntityName = "Battery";
+
+        AZ::IO::Path engineRootPath;
+        m_settingsRegistryInterface->Get(engineRootPath.Native(), AZ::SettingsRegistryMergeUtils::FilePathKey_EngineRootFolder);
+        AZ::IO::Path carPrefabFilepath = engineRootPath / carPrefabName;
+        AZ::IO::Path wheelPrefabFilepath = engineRootPath / wheelPrefabName;
+
+        AZ::EntityId engineEntityId = CreateEditorEntityUnderRoot(engineEntityName);
+        AZ::EntityId tireEntityId = CreateEditorEntityUnderRoot(tireEntityName);
+        AZ::EntityId wheelContainerId = CreateEditorPrefab(wheelPrefabFilepath, { tireEntityId });
+        AZ::EntityId batteryEntityId = CreateEditorEntityUnderRoot(batteryEntityName);
+        AZ::EntityId carContainerId = CreateEditorPrefab(carPrefabFilepath, { engineEntityId, wheelContainerId, batteryEntityId });
+
+        // Validate child entity order under car before detaching the car prefab.
```

Comment:
This part is failing because entity order issue in create-prefab. It should pass once this PR https://github.com/o3de/o3de/pull/14184 gets merged.